### PR TITLE
Change to qsr unique id shorts + new way of registering QSRs in QSRlib

### LINKED ISSUE #101: [QSRs] Drop `qsr_type` and refactor `qsr_keys` to `unique_id` (or smth in these lines)
Refactor `qsr_type` to `longname` and `qsr_keys` to `shortname`.

### PR DESCRIPTION
**Pull request by yianni**
- Removed `<QSR>.qsr_type`, longnames no longer available
- Renamed `<QSR>.qsr_keys` to `<QSR>.unique_id`; this uses the shortnames
- Changed the way that QSRs are registered to QSRlib; provides checking of uniqueness for QSRs unique ids and class names
- Updated ros_example_client to reflect these changes
- Fixed super.__init__ in remaining children
- Updated argprobd to comply with this PR
- Resolved conflicts

Closes #101 #108

### CODE REVIEW

File: qsr_lib/src/qsrlib/qsrlib.py [hunk 1]
```diff
@@ -88,35 +88,45 @@ class QSRlib(object):
...
+                         ("rcc3", QSR_RCC3_Rectangle_Bounding_Boxes_2D),
+                         ("rcc8", QSR_RCC8_Rectangle_Bounding_Boxes_2D),
+                         ("coneDir", QSR_Cone_Direction_Bounding_Boxes_Centroid_2D),
+                         ("qtcbs", QSR_QTC_B_Simplified),
+                         ("qtccs", QSR_QTC_C_Simplified),
+                         ("qtcbcs", QSR_QTC_BC_Simplified),
+                         ("argd", QSR_Arg_Relations_Distance),
+                         ("mos", QSR_Moving_or_Stationary))
+
+        self.__qsrs = self.check_and_activate_qsrs(qsrs_registry)
```
Comment (cdondrup) [inline qsr_lib/src/qsrlib/qsrlib.py]: I still don't like that you have to specify the `unique_id` again inside each tuple. If the `unique_id` would be moved out of the init function and therefore made static, you could just do:

```
(QSR_RCC2_Rectangle_Bounding_Boxes_2D.unique_id, QSR_RCC2_Rectangle_Bounding_Boxes_2D)
```

Or even better just have a list of classes and use the `unique_id` in the new `check_and_activate_qsrs` function.

**Comment by cdondrup** [comment]
I like this, but it is quite a big change and is not backwards compatible. So all code using the library will have to be changed after this is merged and should therefore be communicated well.

File: qsr_lib/src/qsrlib/qsrlib.py (hunk 1 above)
Comment (yianni) [inline qsr_lib/src/qsrlib/qsrlib.py]: Good idea. Let me change it and will push a commit.

**Comment by yianni** [comment]
> I like this, but it is quite a big change and is not backwards compatible. So all code using the library will have to be changed after this is merged and should therefore be communicated well.

I know (hence the danger label, maybe we need a non-backward also :)). I could make it backward compatible if we decide so. Opted out to not to though as to get rid the long names once and for all.

**Comment by cdondrup** [comment]
+1 for loosing all the long lables. If that means that it is not backwards compatible, so be it. Just saying it has to be communicated and the correct version number should be bumped for a re-release. So I vote for doing it (still have to test it though) but maybe we should give it a few days and make sure everybody using the lib sees this.

File: qsr_lib/src/qsrlib/qsrlib.py (hunk 1 above)
Comment (cdondrup) [inline qsr_lib/src/qsrlib/qsrlib.py]: Thanks!

**Comment by yianni** [comment]
Agree. Btw, ouf haven't made `unique_id` pseudostatic as `__unique_id`. Let me do that also and repush. Have to go through all the files again :(.

**Comment by yianni** [comment]
Test and merge please.

**Comment by yianni** [comment]
Ah yes leave a few day for people to know. Where can someone make repo annoucenments?

**Comment by cdondrup** [comment]
WP8 mailing list?!

**Comment by yianni** [comment]
Cheers. Sent an email to the WP8 list. I will be away next week so if you want merge some time by the end of next week/beginning of the other so if there is anything still broken I can fix it when I am back.

**Comment by cdondrup** [comment]
Works for me, opened a PR against your repo with a fix for the qtc tests.

**Comment by yianni** [comment]
Will resolve the conflict. Suspect cause of PR #117

**Comment by yianni** [comment]
Got to go to a meeting now. Will do later.

**Comment by yianni** [comment]
@cdondrup Blocking this till #120 is resolved.

**Comment by cdondrup** [comment]
After #124 is closed, I'm going to merge this tomorrow and start rebasing my current PR.

@pet1330 can then also work on #111

**Comment by cdondrup** [comment]
Since #120 and #124 have been closed, I'm going to merge this now, so work on the other issues can continue.

**Pull request merged**